joinmarket/wallet.py

```diff
@@ -177,11 +177,16 @@ def read_wallet_file_data(self, filename, pwd=None):
                 self.index_cache += [[0,0]] * (
                     self.max_mix_depth - len(self.index_cache))
         decrypted = False
+        trieddefault = False
         while not decrypted:
-            if pwd:
-                password = pwd
+            if trieddefault:
+                password = ''
+                trieddefault = True
```

Comment:
Yes, after testing with null and non-null password wallets, the functionality seems to be correct with that single edit.